subprojects/gradle/gradle-logger/src/main/kotlin/com/avito/logger/LoggerRegistry.kt

```diff
@@ -0,0 +1,103 @@
+package com.avito.logger
+
+import com.avito.android.elastic.ElasticConfig
+import com.avito.android.sentry.SentryConfig
+import com.avito.logger.destination.ElasticDestination
+import com.avito.logger.destination.SentryDestination
+import com.avito.logger.destination.Slf4jDestination
+import com.avito.logger.formatter.AppendMetadataFormatter
+import com.avito.logger.handler.CombinedHandler
+import com.avito.logger.handler.DefaultLoggingHandler
+import com.avito.logger.handler.LoggingHandler
+
+object LoggerRegistry {
+
+    private val loggersCache = mutableMapOf<String, Logger>()
+
+    fun create(
+        isCiRun: Boolean,
+        className: String,
+        projectPath: String,
+        sentryConfig: SentryConfig,
+        elasticConfig: ElasticConfig,
+        pluginName: String? = null,
+        taskName: String? = null
+    ): Logger = provideLogger(
+        isCiRun = isCiRun,
+        sentryConfig = sentryConfig,
+        elasticConfig = elasticConfig,
+        metadata = LoggerMetadata(
+            className = className,
+            pluginName = pluginName,
+            projectPath = projectPath,
+            taskName = taskName
+        )
+    )
+
+    private fun provideLogger(
+        isCiRun: Boolean,
+        sentryConfig: SentryConfig,
+        elasticConfig: ElasticConfig,
+        metadata: LoggerMetadata
+    ): Logger = if (isCiRun) {
+        loggersCache.getOrPut(metadata.className) {
+            createCiLogger(sentryConfig, elasticConfig, metadata)
+        }
+    } else {
+        createLocalBuildLogger(metadata)
+    }
+
+    private fun createCiLogger(
+        sentryConfig: SentryConfig,
+        elasticConfig: ElasticConfig,
+        metadata: LoggerMetadata
+    ): Logger {
+
+        val sentryHandler = createSentryHandler(sentryConfig, metadata)
+
+        return DefaultLogger(
+            debugHandler = createElasticHandler(elasticConfig, level = "DEBUG", metadata = metadata),
+            infoHandler = createElasticHandler(elasticConfig, level = "INFO", metadata = metadata),
+            warningHandler = createElasticHandler(elasticConfig, level = "WARNING", metadata = metadata),
+            criticalHandler = CombinedHandler(
+                handlers = listOf(
+                    createElasticHandler(elasticConfig, level = "ERROR", metadata = metadata),
+                    sentryHandler
+                )
+            )
+        )
+    }
+
+    private fun createLocalBuildLogger(metadata: LoggerMetadata): Logger {
+
+        val gradleLoggerHandler = DefaultLoggingHandler(
+            formatter = AppendMetadataFormatter(metadata),
+            destination = Slf4jDestination(metadata.className)
```

Comment:
Gradle creates Logger for tasks by `Task` interface name. Maybe it's an important semantic.

Reply:
i don't know for sure, and don't know what we could achieve with that change; it's not free though, because we will need additional parameter to distinguish between task logger and anything else